File: examples/src/main/java/Hello_PIDE.java
import java.nio.file.Paths;
import edu.tum.cs.isabelle.api.*;
import edu.tum.cs.isabelle.japi.*;
import edu.tum.cs.isabelle.setup.*;

public class Hello_PIDE {

  public static void main(String args[]) {
    Environment env = JSetup.makeEnvironment(Paths.get(System.getenv("ISABELLE_HOME")), JPlatform.guess(), new Version("2015"));
    Configuration config = Configuration.fromPath(Paths.get("."), "Protocol2015");
    JSystem sys = JSystem.create(env, config);
    System.out.println(sys.invoke(Operations.HELLO, "world"));
    sys.dispose();
  }

}

File: examples/src/main/scala/Hello_PIDE.scala
package edu.tum.cs.isabelle.examples.scala

import scala.concurrent._
import scala.concurrent.duration._
import scala.concurrent.ExecutionContext.Implicits.global

import edu.tum.cs.isabelle._
import edu.tum.cs.isabelle.api._
import edu.tum.cs.isabelle.setup._

object Hello_PIDE extends App {


  val transaction =
    for {
      setup <- Setup.defaultSetup(Version("2015")).toOption.get // yolo
      env <- setup.makeEnvironment
      resources = Resources.dumpIsabelleResources()
      config = resources.makeConfiguration(Nil, "Protocol")
      sys <- System.create(env, config)
      response <- sys.invoke(Operation.Hello)("world")
      _ = println(response.unsafeGet)
      () <- sys.dispose
    } yield ()

  Await.result(transaction, Duration.Inf)

}
